staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -830,9 +835,21 @@ func (s *Controller) lockedUpdateLoadBalancerHosts(service *v1.Service, hosts []
 	return err
 }
 
-func wantsLoadBalancer(service *v1.Service) bool {
-	// if LoadBalancerClass is set, the user does not want the default cloud-provider Load Balancer
-	return service.Spec.Type == v1.ServiceTypeLoadBalancer && service.Spec.LoadBalancerClass == nil
+func wantsLoadBalancer(service *v1.Service, allowedLoadbalancerClass []*string) bool {
+	if service.Spec.Type == v1.ServiceTypeLoadBalancer {
+		if service.Spec.LoadBalancerClass == nil {
+			// Default provider should accept svc, and other provider should ignore
+			return len(allowedLoadbalancerClass) == 0
+		}
+
+		// if LoadBalancerClass is set, the user does not want the default cloud-provider Load Balancer
+		for _, className := range allowedLoadbalancerClass {
+			if strings.EqualFold(*className, *service.Spec.LoadBalancerClass) {
```

Comment:
If we found the supported LoadBalancerClass, shouldn't we return `false` for cloud providers? Since users indicate that they don't want the cloud providers' load balancer implementation and we also found that the implementation they want to use is also supported.

Reply:
wantloadbalancer return true only when svc is accpeted by current load balancer implementation. The svc is accepted only when svc meets LoadBalancerClass requirement set by loadbalancer implementation. 
If this controller is launched by non-default cloud provider, here it should return true.